src/x/xHilbertTransform.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Performs the Hilbert transform
+ * @returns A new vector with 90 degree shift regarding the phase of the original function
+ */
+
+export function hilbertTransform(
+  input: number[],
+  options: { inClockwise?: boolean } = {},
+) {
+  const { inClockwise = true } = options;
+  const array = [0, 0].concat(input).concat(0);
```

Comment:
Please use: `[0, 0, ...input, 0);`

It prevents to duplicate the array twice.

Reply:
Could you check to be sure that this first element is really used in your code. Put a NaN for example to test.

Reply:
Those zeros are trying to minimize the edges effect when the input array is too short. If the input array has reasonable length (at least 256 point), the edges effect is minimum and those zeros are useless.

It will crush because line 12, but indeed there is a zero more.

Reply:
You didn't use the spread operator yet. Also I tried to change the first element to NaN and the test are still passing. Are you sure this first '0' is really usefull and being used ?